### PR TITLE
Updated the DiscountCodeValidation model.

### PR DESCRIPTION
**Pull request by YevhenShBolt**
- Moved a coupon applying process to the separate method
- Added the gift card applying process, also placed in a separate method.
- Fixed the returning statements for the errors.
- Fixed a message returning an error to the Bugsnag in exception

### CODE REVIEW

File: Model/Api/DiscountCodeValidation.php [hunk 1]
```diff
@@ -444,21 +538,27 @@ private function sendErrorResponse($errCode, $message, $httpStatusCode, $quote =
...
      * @param array $result
      * @param Quote $quote
+     * @return array
      * @throws \Exception
      */
-    private function sendSuccessResponse($result, $quote) {
+    private function sendSuccessResponse($result, $quote)
+    {
         $result['cart'] = $this->getCartTotals($quote);
+        $this->bugsnag->notifyError('Discount Validation Response', json_encode($result));
```
Comment (daisy1754) [inline Model/Api/DiscountCodeValidation.php]: I'm sure it's for debugging; please make sure you remove it

File: Model/Api/DiscountCodeValidation.php [hunk 2]
```diff
@@ -288,130 +295,213 @@ public function validate() {
...
-                        \IntlDateFormatter::MEDIUM
-                    );
-                return $this->sendErrorResponse(
-                    self::ERR_CODE_NOT_AVAILABLE,
-                    $desc,
-                    422,
-                    $immutableQuote
-                );
-            }
+            return $this->sendSuccessResponse($result, $immutableQuote);
```
Comment (daisy1754) [inline Model/Api/DiscountCodeValidation.php]: applyingCouponCode method may call sendErrorResponse method internally. I guess in such case we shouldn't call sendSuccessResponse?

File: Model/Api/DiscountCodeValidation.php (hunk 1 above)
Comment (YevhenShBolt) [reply Model/Api/DiscountCodeValidation.php]: I was sure that I removed it.

File: Model/Api/DiscountCodeValidation.php (hunk 2 above)
Comment (YevhenShBolt) [reply Model/Api/DiscountCodeValidation.php]: Yes sure, i will think and refactor it.

Comment (YevhenShBolt) [reply Model/Api/DiscountCodeValidation.php]: I've added check above

File: etc/module.xml [hunk 3]
```diff
@@ -24,6 +24,7 @@
             <module name="Magento_Sales"/>
             <module name="Magento_Checkout"/>
             <module name="Magento_Tax"/>
+            <module name="Magento_GiftCardAccount"/>
```
Comment (danac-gs) [inline etc/module.xml]: What happens here if the module does not exist? Assuming it is ignored gracefully.

Comment (YevhenShBolt) [reply etc/module.xml]: Nothing, i checked. It just determines (or like) the order of module load.

**Pull request merged**